libsolidity/formal/SMTChecker.cpp

```diff
@@ -383,17 +383,43 @@ void SMTChecker::endVisit(Assignment const& _assignment)
 
 void SMTChecker::endVisit(TupleExpression const& _tuple)
 {
-	if (
-		_tuple.isInlineArray() ||
-		_tuple.components().size() != 1 ||
-		!isSupportedType(_tuple.components()[0]->annotation().type->category())
-	)
+	if (_tuple.isInlineArray())
 		m_errorReporter.warning(
 			_tuple.location(),
-			"Assertion checker does not yet implement tuples and inline arrays."
+			"Assertion checker does not yet implement inline arrays."
 		);
+	else if (_tuple.annotation().type->category() == Type::Category::Tuple)
+	{
+		createExpr(_tuple);
+		vector<shared_ptr<SymbolicVariable>> components;
+		for (auto const& component: _tuple.components())
+		{
+			if (component)
+			{
+				if (auto varDecl = identifierToVariable(*component))
```

Comment:
Why do you have to make this distinction?

Reply:
To know where to read the `SymbolicVariable` from: `m_variables` or `m_expressions`. If we just needed the `smt::Expression` we could just get `expr(*component)` which would read from the `Identifier` expression, but in this case we need the variable declaration's `SymbolicVariable` in case this is the left hand side of an assignment.

Reply:
Ah I see!